app/src/main/java/com/example/bottomnav/UnsignedIntegerDecoder.java

```diff
@@ -1,25 +1,18 @@
 package com.example.bottomnav;
 
 import java.math.BigInteger;
-import java.nio.ByteBuffer;
+import java.util.Optional;
 
 public class UnsignedIntegerDecoder<T> extends IntegerDecoder{
     int sign;
 
-    public UnsignedIntegerDecoder(VariableContents contents, int givenSign) {
-        super(contents);
+    public UnsignedIntegerDecoder(int size, int givenSign, String name) {
```

Comment:
"givenSign" is used for bitwise operation when trying to decode the payload as unsigned. I can't really use a universal hexadecimal number because doing bitwise op. needs both the payload and "givenSign" to be the same length.

Reply:
I don't think you can use a universal hex number, but you could derive the hex number from the byte size right?
For example, size=1 gives `0xff`, size=2 gives `0xffff`, size=4 (invalid here) gives `0xffffffff`, and in general, does `givenSign=1 << (8 * size) - 1` work? (note, `1 << (8 * size)` is the bit shift version of 2 ^ (8 * size))